`tests/misc/llext-edk/misc.edk.pytest` build failure when running under load

<!--
**Notes**
Github Discussions (https://github.com/zephyrproject-rtos/zephyr/discussions)
are available to first verify that the issue is a genuine Zephyr bug and not a
consequence of Zephyr services misuse.

This issue list is only for bugs in the main Zephyr code base
(https://github.com/zephyrproject-rtos/zephyr/). If the bug is for a project
fork (such as NCS) specific feature, please open an issue in the fork project
instead.
-->

**Describe the bug**

While tests are batched run and the system's under heavy load, `tests/misc/llext-edk/misc.edk.pytest` fails with the following error:


```
-------------------------------- live log call ---------------------------------
INFO: -- west build: running target llext-edk
[1/1] Generating llext-edk.tar.xz
-- Found Python3: /usr/bin/python (found suitable version "3.10.12", minimum required is "3.10") found components: Interpreter
-- Cache files will be written to: /root/.cache/zephyr
DEBUG: Copying llext-edk.tar.xz to /tmp/tmps7cievp7
DEBUG: Extracting llext-edk.tar.xz to /tmp/tmps7cievp7
INFO:
DEBUG: Copying extension to /tmp/tmprpbpnrf2
DEBUG: Copying file2hex.py to /tmp/tmprpbpnrf2
DEBUG: Building extension in /tmp/tmprpbpnrf2 - cmake
INFO: -- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/tmprpbpnrf2/build
DEBUG: Building extension in /tmp/tmprpbpnrf2 - make
make: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
/tmp/tmprpbpnrf2/src/main.c:7:10: fatal error: zephyr/kernel.h: No such file or directory
7 | #include <zephyr/kernel.h>
|          ^~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/extension.dir/build.make:73: extension.llext] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/extension.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
FAILED
```

but if we run this test alone (and on a fresh out directory)
`west twister -p qemu_cortex_r5/zynqmp_rpu -s tests/misc/llext-edk/misc.edk.pytest` it passes.

If the test has failed, trying to re-run whist preserving the output directory will keep on failing. The output directory (for that specific test is enough) needs to be removed in order to get the tests to build from fresh & pass.

 - What target platform are you using?
   Was noticed with `qemu_cortex_r5/zynqmp_rpu`. I don't know if there are other platforms running this specific test.
 - What have you tried to diagnose or workaround this issue?
   Clearing the tests output directory and running this test alone. It passed.
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
   It might be but considering how long it takes to run the suites and trigger the issue, a bisect hasn't be an option.

**To Reproduce**

Run `west twister -p qemu_cortex_r5/zynqmp_rpu`

**Expected behavior**

The test should build and pass on the first run.

**Impact**

It is pretty significant as it prevents local & CI run to pass.

**Logs and console output**
<!--
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)
-->

**Environment (please complete the following information):**

 - OS: Linux (the CI docker container 0.27.4)
 - Commit SHA: caa284e0355

--- a/tests/misc/llext-edk/testcase.yaml
+++ b/tests/misc/llext-edk/testcase.yaml
@@ -7,3 +7,4 @@ tests:
     platform_allow:
       - qemu_cortex_r5
     toolchain_exclude: llvm
+    timeout: 120 # See #85662

